### PR TITLE
[Release] 0.10.0 Premature

### PR DESCRIPTION
**Pull request by ruseinov**
## Summary of changes

Changes introduced in this pull request:

- Preparing for the release

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/3032

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: documentation/src/developer_documentation/release_checklist.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ in case of absence, passing the task to a different team member.
 4. Once the changes in step 1 and step 2 have been merged, tag the commit with
    the new version number. The version tag should start with a lowercase 'v'.
    Example: v0.4.1
-5. (NOT YET APPLICABLE) Publish new crates on crates.io.
+5. Publish new crates on crates.io.
```
Comment (LesnyRumcajs) [inline documentation/src/developer_documentation/release_checklist.md:23]: This description should be extended. We can do it in a separate PR, though.

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +35,15 @@
 
 ### Fixed
 
+## Forest v0.10.0 "Premature"
+
+### Breaking
+
+- [#3007](https://github.com/ChainSafe/forest/pull/3007): Optimize DB
+  parameters. This requires all existing databases to be re-initialized.
+
```
Comment (LesnyRumcajs) [inline CHANGELOG.md:44]: Why is it in the diff? It should already be in the `main`, no?

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — COMMENTED** [review]
Are there any changes to the lockfile?

File: CHANGELOG.md (hunk 2 above)
Comment (lemmih) [reply CHANGELOG.md:44]: It was moved from `Unreleased` to `Forest v0.10.0 "Premature"`

File: documentation/src/developer_documentation/release_checklist.md (hunk 1 above)
Comment (ruseinov) [reply documentation/src/developer_documentation/release_checklist.md:23]: I'm not sure if it's worth duplicating what's already written [here](https://doc.rust-lang.org/cargo/reference/publishing.html). I'll just link it.

File: CHANGELOG.md (hunk 2 above)
Comment (LesnyRumcajs) [reply CHANGELOG.md:47]: Ah, my bad, I expected the diff to show it differently.

File: documentation/src/developer_documentation/release_checklist.md (hunk 1 above)
Comment (LesnyRumcajs) [reply documentation/src/developer_documentation/release_checklist.md]: Let me rephrase - which crates? If there's just the `forest-filecoin`, then let's use the singular form.

**Review by LesnyRumcajs — APPROVED** [review]

File: documentation/src/developer_documentation/release_checklist.md (hunk 1 above)
Comment (ruseinov) [reply documentation/src/developer_documentation/release_checklist.md]: fixed in another PR #3040
